6.5.group_by.py

```diff
@@ -0,0 +1,14 @@
+from collections import defaultdict
+
+
+def group_by(f, iterable):
+    res = defaultdict(list)
+    for i in iterable:
+        key = f(i)
+        res[key].append(i)
+    return dict(res)
+
+
+
+# a = group_by(len, ["hi", "bye", "yo", "try", "oooo", "p"])

```

Comment:
You should not leave a documented code in your scripts.
Better put it in a test function.